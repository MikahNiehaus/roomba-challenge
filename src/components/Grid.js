import React, { Component } from 'react';
import Tile from './Tile';

class Grid extends Component {

    printGrid = (array) => {
        return array.map((row) => {
            return <tr>
                {row}
            </tr>
        });
    };

    render() {
        let count = Math.floor((Math.random() * 100000) + 1);
        const grid = this.props.state.grid;
        const width = grid.width;
        const length = grid.length;
        const arrayOfTiles = [];

        for (let i = length - 1; i >= 0; i--) {
            const row = [];

            for (let b = 0; b < width; b++) {
                row.push(
                        <td>
                            <Tile
                                x={b}
                                y={i}
                                dirt={this.props.state.dirt}
                                roomba={this.props.state.roomba}
                                key={count}
                            />
                        </td>
                );

                count++;
            }

            arrayOfTiles.push(row);
        }

        return (
            <div>
                <table>
                    <tbody>
                        {this.printGrid(arrayOfTiles)}
                    </tbody>
                </table>
            </div>
        )
    }
}

export default Grid;