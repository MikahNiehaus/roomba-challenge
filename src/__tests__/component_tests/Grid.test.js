import React from 'react';
import {shallow} from 'enzyme';
import Grid from '../../components/Grid';
import sinon from 'sinon';

let grid;

describe('Grid', () => {

    let splitUpTilesStub;

    beforeEach(() => {
        const props = {
            state: {
                grid: {
                    length: 2,
                    width: 2
                },
            }
        };

        grid = shallow(<Grid {...props}/>);
    });

    it('renders correctly', () => {
        expect(grid).toMatchSnapshot();
    });

    describe('splitUpTiles', () => {
        it('returns an array (a) that is split up into (l) number of arrays', () => {
            const array = [{x:2, y:2}, {x:1, y:1}, {x:4, y:0}, {x:2, y:1}];
            const l = 2;

            expect(grid.instance().splitUpTiles(array, l)).toEqual([[{x:2, y:2}, {x:1, y:1}], [{x:4, y:0}, {x:2, y:1}]])
        });
    });

    describe('createColumns', () => {
        it('returns an array of data wrapped in table rows', () => {
            const splitUpTilesArrayOfArrays = [[{"x": 4, "y": 0}, {"x": 2, "y": 1}, {"x": 2, "y": 2}, {"x": 1, "y": 1}]];

            splitUpTilesStub = sinon.stub(grid.instance(), "splitUpTiles");
            splitUpTilesStub.returns(splitUpTilesArrayOfArrays);
            //expect(grid.instance().createColumns()).toEqual('[<tr>{"x": 4, "y": 0}{"x": 2, "y": 1}{"x": 2, "y": 2}{"x": 1, "y": 1}</tr>, <tr />]')
        });
    });

});