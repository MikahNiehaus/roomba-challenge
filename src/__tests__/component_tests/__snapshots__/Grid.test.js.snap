// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Grid
    state={
        Object {
            "grid": Object {
              "length": 2,
              "width": 2,
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table>
        <tbody>
                <tr>
                        <td>
                                <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={0}
                                        y={1}
                                />
                        </td>
                        <td>
                                <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={1}
                                        y={1}
                                />
                        </td>
                </tr>
                <tr>
                        <td>
                                <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={0}
                                        y={0}
                                />
                        </td>
                        <td>
                                <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={1}
                                        y={0}
                                />
                        </td>
                </tr>
        </tbody>
</table>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tbody>
          <tr>
                    <td>
                              <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={0}
                                        y={1}
                              />
                    </td>
                    <td>
                              <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={1}
                                        y={1}
                              />
                    </td>
          </tr>
          <tr>
                    <td>
                              <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={0}
                                        y={0}
                              />
                    </td>
                    <td>
                              <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={1}
                                        y={0}
                              />
                    </td>
          </tr>
</tbody>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <td>
                            <Tile
                                          dirt={undefined}
                                          roomba={undefined}
                                          x={0}
                                          y={1}
                            />
              </td>
              <td>
                            <Tile
                                          dirt={undefined}
                                          roomba={undefined}
                                          x={1}
                                          y={1}
                            />
              </td>
</tr>,
            <tr>
              <td>
                            <Tile
                                          dirt={undefined}
                                          roomba={undefined}
                                          x={0}
                                          y={0}
                            />
              </td>
              <td>
                            <Tile
                                          dirt={undefined}
                                          roomba={undefined}
                                          x={1}
                                          y={0}
                            />
              </td>
</tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                  <Tile
                                    dirt={undefined}
                                    roomba={undefined}
                                    x={0}
                                    y={1}
                  />
</td>,
                <td>
                  <Tile
                                    dirt={undefined}
                                    roomba={undefined}
                                    x={1}
                                    y={1}
                  />
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Tile
                    dirt={undefined}
                    roomba={undefined}
                    x={0}
                    y={1}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "94885",
                  "nodeType": "class",
                  "props": Object {
                    "dirt": undefined,
                    "roomba": undefined,
                    "x": 0,
                    "y": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Tile
                    dirt={undefined}
                    roomba={undefined}
                    x={1}
                    y={1}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "94886",
                  "nodeType": "class",
                  "props": Object {
                    "dirt": undefined,
                    "roomba": undefined,
                    "x": 1,
                    "y": 1,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                  <Tile
                                    dirt={undefined}
                                    roomba={undefined}
                                    x={0}
                                    y={0}
                  />
</td>,
                <td>
                  <Tile
                                    dirt={undefined}
                                    roomba={undefined}
                                    x={1}
                                    y={0}
                  />
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Tile
                    dirt={undefined}
                    roomba={undefined}
                    x={0}
                    y={0}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "94887",
                  "nodeType": "class",
                  "props": Object {
                    "dirt": undefined,
                    "roomba": undefined,
                    "x": 0,
                    "y": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Tile
                    dirt={undefined}
                    roomba={undefined}
                    x={1}
                    y={0}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "94888",
                  "nodeType": "class",
                  "props": Object {
                    "dirt": undefined,
                    "roomba": undefined,
                    "x": 1,
                    "y": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table>
          <tbody>
                    <tr>
                              <td>
                                        <Tile
                                                  dirt={undefined}
                                                  roomba={undefined}
                                                  x={0}
                                                  y={1}
                                        />
                              </td>
                              <td>
                                        <Tile
                                                  dirt={undefined}
                                                  roomba={undefined}
                                                  x={1}
                                                  y={1}
                                        />
                              </td>
                    </tr>
                    <tr>
                              <td>
                                        <Tile
                                                  dirt={undefined}
                                                  roomba={undefined}
                                                  x={0}
                                                  y={0}
                                        />
                              </td>
                              <td>
                                        <Tile
                                                  dirt={undefined}
                                                  roomba={undefined}
                                                  x={1}
                                                  y={0}
                                        />
                              </td>
                    </tr>
          </tbody>
</table>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tbody>
            <tr>
                        <td>
                                    <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={0}
                                                y={1}
                                    />
                        </td>
                        <td>
                                    <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={1}
                                                y={1}
                                    />
                        </td>
            </tr>
            <tr>
                        <td>
                                    <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={0}
                                                y={0}
                                    />
                        </td>
                        <td>
                                    <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={1}
                                                y={0}
                                    />
                        </td>
            </tr>
</tbody>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                                <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={0}
                                                y={1}
                                />
                </td>
                <td>
                                <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={1}
                                                y={1}
                                />
                </td>
</tr>,
              <tr>
                <td>
                                <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={0}
                                                y={0}
                                />
                </td>
                <td>
                                <Tile
                                                dirt={undefined}
                                                roomba={undefined}
                                                x={1}
                                                y={0}
                                />
                </td>
</tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={0}
                                        y={1}
                    />
</td>,
                  <td>
                    <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={1}
                                        y={1}
                    />
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tile
                      dirt={undefined}
                      roomba={undefined}
                      x={0}
                      y={1}
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "94885",
                    "nodeType": "class",
                    "props": Object {
                      "dirt": undefined,
                      "roomba": undefined,
                      "x": 0,
                      "y": 1,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tile
                      dirt={undefined}
                      roomba={undefined}
                      x={1}
                      y={1}
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "94886",
                    "nodeType": "class",
                    "props": Object {
                      "dirt": undefined,
                      "roomba": undefined,
                      "x": 1,
                      "y": 1,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                    <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={0}
                                        y={0}
                    />
</td>,
                  <td>
                    <Tile
                                        dirt={undefined}
                                        roomba={undefined}
                                        x={1}
                                        y={0}
                    />
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tile
                      dirt={undefined}
                      roomba={undefined}
                      x={0}
                      y={0}
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "94887",
                    "nodeType": "class",
                    "props": Object {
                      "dirt": undefined,
                      "roomba": undefined,
                      "x": 0,
                      "y": 0,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Tile
                      dirt={undefined}
                      roomba={undefined}
                      x={1}
                      y={0}
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": "94888",
                    "nodeType": "class",
                    "props": Object {
                      "dirt": undefined,
                      "roomba": undefined,
                      "x": 1,
                      "y": 0,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
